src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -125,5 +150,34 @@ private static bool TryFindFurthestNearbyComment<T>(ref T triviaSearchList, out
 
             return nearbyTrivia.IsSingleOrMultiLineComment();
         }
+
+        private static QuickInfoItem? BuildQuickInfoDirectives(SyntaxToken token, CancellationToken cancellationToken)
+        {
+            var directiveTrivia = token.Parent?.AncestorsAndSelf(ascendOutOfTrivia: false).FirstOrDefault(n =>
+                n is EndRegionDirectiveTriviaSyntax or ElifDirectiveTriviaSyntax or ElseDirectiveTriviaSyntax or EndIfDirectiveTriviaSyntax) as DirectiveTriviaSyntax;
+            if (directiveTrivia is EndRegionDirectiveTriviaSyntax endRegionDirectiveTrivia)
+            {
+                var regionStart = endRegionDirectiveTrivia.GetMatchingDirective(cancellationToken);
+                if (regionStart is not null)
+                {
+                    return QuickInfoItem.Create(token.Span, relatedSpans: ImmutableArray.Create(regionStart.Span));
+                }
+            }
+
+            if (directiveTrivia is ElifDirectiveTriviaSyntax or ElseDirectiveTriviaSyntax or EndIfDirectiveTriviaSyntax)
+            {
+                var matchingDirectives = directiveTrivia.GetMatchingConditionalDirectives(cancellationToken);
+                var lastMatchBefore = matchingDirectives
+                    .TakeWhile(d => d.SpanStart < directiveTrivia.SpanStart)
+                    .Select(d => d.Span)
+                    .ToImmutableArray();
+                if (lastMatchBefore.Length > 0)
```

Comment:
this is named weird.  it says 'lastMatchBefore'.  but it matches multiple things.